source/framework/vx_graph_verify.c

```diff
@@ -730,98 +727,152 @@ static vx_status ownGraphCalcInAndOutNodes(vx_graph graph)
+                                else if ((uint32_t)VX_INPUT == prm_next_dir)
+                                {
+                                    ++inputs_attached;
                                     /* add node_next as output node for current node if not already added */
-                                    status = ownNodeAddOutNode(node_cur, node_next);
-
-                                    if(status == (vx_status)VX_SUCCESS)
+                                    /* but we don't do it if this parameter is an output attached to a bidirectional */
+                                    if (0U == biputs_attached)
                                     {
-                                        /* add node_current as input node for next node if not already added */
-                                        status = ownNodeAddInNode(node_next, node_cur);
-                                        if (status != (vx_status)VX_SUCCESS)
+                                        status = ownNodeAddOutNode(node_cur, node_next);
+                                        if(status == (vx_status)VX_SUCCESS)
                                         {
-                                            VX_PRINT(VX_ZONE_ERROR,"Add in node at index %d failed\n", node_cur_idx);
+                                            /* add node_current as input node for next node if not already added */
+                                            status = ownNodeAddInNode(node_next, node_cur);
+                                            if (status != (vx_status)VX_SUCCESS)
+                                            {
+                                                VX_PRINT(VX_ZONE_ERROR,"Add in node at index %d failed\n", node_cur_idx);
+                                            }
+                                        }
+                                        else
+                                        {
+                                            VX_PRINT(VX_ZONE_ERROR,"Add out node at index %d failed\n", node_cur_idx);
                                         }
                                     }
+                                }
+                                else if ((vx_enum)VX_BIDIRECTIONAL == prm_next_dir)
+                                {
+                                    /* Current node must be an output, proceed as if it was the only input attached */
+                                    /* We must make the bidirectional parameter node the only one following the
+                                       output parameter node; to do this we go through all the out nodes and replace
+                                       the output parameter node by the bidirectional parameter node in the in node list */
+                                    ++biputs_attached;
+                                    uint32_t i;
+                                    for (i = 0; i < ownNodeGetNumOutNodes(node_cur) && ((vx_status)VX_SUCCESS == status); ++i)
```

Comment:
I am still getting the below error here.  You may need to add further parentheses

"The precedence of operators within expressions should be made explicit"